libs/auth/testing/src/drivers/testing/register/register.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable, Inject } from '@angular/core';
+import { Observable, of } from 'rxjs';
+
+import {
+  DaffAccountRegistration,
+  DaffAuthToken,
+  DaffRegisterServiceInterface,
+  DaffCustomerRegistration,
+} from '@daffodil/auth';
+
+import { DaffAuthTokenFactory } from '../../../factories/auth-token.factory';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffTestingRegisterService implements DaffRegisterServiceInterface<
+  DaffAccountRegistration<DaffCustomerRegistration>,
+  DaffCustomerRegistration,
+  DaffAuthToken
+> {
+  constructor (@Inject(DaffAuthTokenFactory) private factory: DaffAuthTokenFactory) {}
```

Comment:
Since this is an `@Injectable` you don't need the `@Inject`

Reply:
🤔 Clearly I misunderstand something about Angular DI. How does Angular know what class to inject for `DaffAuthTokenFactory`? Is it the `providedIn: 'root'` option on the factory `@Injectable` decorator that does it?

Reply:
In other instances, (where we use injection tokens) we're relying on an interface that doesn't exist in `js`-land, so we're forced to use an InjectionToken. In this instance, the class name exists in `js`-land, and combined with the `providedIn: 'root'` this providing is handled by the compiler IFF the token is used in the compilation (otherwise it gets tree-shaken out).